### PR TITLE
Fix broken test for models with batchnorm

### PR DESCRIPTION
**Pull request by Rocketknight1**
One of the Keras tests assumed that fitting a model for one iteration with a learning rate of zero would not change any weights. This is not true for `BatchNorm`, which updates its running means and variances regardless! As a result, the model after the iteration had slightly different outputs, which caused the test to be very flaky. We now reinitialize the model after the single training epoch to make sure this doesn't happen.

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Related to #17427, cc @amyeroberts .

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for the fix! ❤️

**Comment by gante** [comment]
🧠

**Comment by Rocketknight1** [comment]
Give @amyeroberts the brain emoji for that one - she identified the whole problem, I just fixed the test!
